### PR TITLE
Enable default metadata cache for FUSE SDK

### PR DESCRIPTION
**Pull request by LuQQiu**
### What changes are proposed in this pull request?

Enable default client-side metadata cache (20,000 entries with 40MB memory).

### Why are the changes needed?

Each fuse.open() has two extra get file status calls which cannot be provided via kernel metadata cache but can via user space metadata cache. Enable user space metadata cache can largely improve the overall fuse FIO small file read performance. S3FS-FUSE has similar settings of enable user space metadata cache (100,000 entries with 40MB memory) by default.

FIO benchmarking FUSE SDK against s3-fuse in terms of small file read:
fio --name=sequentialread --rw=read --bs=100k --numjobs=1 --filesize=100k --direct=1 --group_reporting --nrfiles 1000

<google-sheets-html-origin><style type="text/css"></style>

  | S3 FUSE | ALLUXIO FUSE
-- | -- | --
Original | 9000 KiB/s | 1500 KiB/s
Kernal metadata cache (attr_timeout=7200, entry_timeout=7200) | 9000 KiB/s | 2400 KiB/s
User metadata cache | 9000 KiB/s | 1150 MiB/s

### Does this PR introduce any user facing changes?

Yeah, alluxio-fuse mount <..> metadata cache option with default value changed
-o metadata_cache_size=<size> (Default=\"20000\" (around 40MB memory)) 
-o metadata_cache_expire=<timeout> (Default=\"no expire\")

### CODE REVIEW

**Comment by LuQQiu** [comment]
@huanghua78 Please take a look to change the default metadata cache settings. Thanks

File: tests/src/test/java/alluxio/client/fuse/AbstractFuseIntegrationTest.java [hunk 1]
```diff
@@ -106,6 +106,7 @@ public static void beforeClass() {
 
   @Before
   public void before() throws Exception {
+    Configuration.set(PropertyKey.USER_METADATA_CACHE_MAX_SIZE, 0);
```
Comment (huanghua78) [inline tests/src/test/java/alluxio/client/fuse/AbstractFuseIntegrationTest.java:109]: One of the case is: PropertyKey.USER_METADATA_CACHE_MAX_SIZE = 0;
What about the other case where PropertyKey.USER_METADATA_CACHE_MAX_SIZE is some other value, e.g. 1000.

File: core/client/fs/src/main/java/alluxio/client/file/FileSystemUtils.java [hunk 2]
```diff
@@ -39,6 +40,14 @@ public final class FileSystemUtils {
   // prevent instantiation
   private FileSystemUtils() {}
 
+  /**
+   * @param conf the configuration to get info from
+   * @return whether metadata cache is enabled in the given configuration
+   */
+  public static boolean metadataEnabled(AlluxioConfiguration conf) {
```
Comment (yyongycy) [inline core/client/fs/src/main/java/alluxio/client/file/FileSystemUtils.java:47]: so far there are several caches, could the name be changed more explicit to indicate the fuse user space cache?

Comment (LuQQiu) [reply core/client/fs/src/main/java/alluxio/client/file/FileSystemUtils.java:47]: There are two metadata cache
- FUSE KERNEL metadata cache enabled by mount option (e.g. -o kernel_cache)
- USER SPACE metadata cache, currently we only have client side metadata cache, defined by this property key.
It's shared by all clients, not just fuse clients, e.g. S3 API can use it, java API can use it, not just POSIX API

File: tests/src/test/java/alluxio/client/fuse/AbstractFuseIntegrationTest.java (hunk 1 above)
Comment (LuQQiu) [reply tests/src/test/java/alluxio/client/fuse/AbstractFuseIntegrationTest.java:109]: Our current Fuse tests focus on the case without any cache.
We do have metadata cache test tests the functionality of metadata  cache, similarly, we have data cache tests covers the functionality of client side data cache.

Good call that in the future can include more fuse tests covering the test cases with data/metadata cache.

If the metadata cache max size is bigger than zero, then metadata cache is enabled.

**Review by yyongycy — APPROVED** [review]
lgtm

File: tests/src/test/java/alluxio/client/fuse/AbstractFuseIntegrationTest.java (hunk 1 above)
Comment (huanghua78) [reply tests/src/test/java/alluxio/client/fuse/AbstractFuseIntegrationTest.java:109]: OK

**Review by huanghua78 — APPROVED** [review]
Looks good.

**Comment by LuQQiu** [comment]
Thanks for reviewing @yyongycy @huanghua78 !

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**